src/m365/spo/commands/commandset/commandset-remove.ts

```diff
@@ -0,0 +1,183 @@
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId as string)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.scope && SpoCommandSetRemoveCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpoCommandSetRemoveCommand.scopes.join(', ')}`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'title', 'clientSideComponentId'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Removing ListView Command Set ${args.options.clientSideComponentId || args.options.title || args.options.id} to site '${args.options.webUrl}'...`);
+    }
+
+    if (args.options.confirm) {
+      await this.deleteCommandset(args);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove command set '${args.options.clientSideComponentId || args.options.title || args.options.id}'?`
+      });
+
+      if (result.continue) {
+        await this.deleteCommandset(args);
+      }
+    }
+  }
+
+  private async getCommandSetId(options: Options): Promise<string> {
+    if (options.id) {
+      return options.id;
+    }
+
+    if (options.title) {
+      const commandSets: CustomAction[] = await spo.getCustomActions(options.webUrl, options.scope, `(Title eq '${formatting.encodeQueryParameter(options.title as string)}') and (startswith(Location,'ClientSideExtension.ListViewCommandSet'))`);
+
+      if (commandSets.length === 0) {
+        throw `No user commandsets with title '${options.title}' found`;
+      }
+
+      if (commandSets.length > 1) {
+        throw `Multiple user commandsets with title '${options.title}' found. Please disambiguate using IDs: ${commandSets.map(a => a.Id).join(', ')}`;
+      }
+
+      return commandSets[0].Id;
+    }
+
+    const commandSets: CustomAction[] = await spo.getCustomActions(options.webUrl, options.scope, `(ClientSideComponentId eq guid'${options.clientSideComponentId}') and (startswith(Location,'ClientSideExtension.ListViewCommandSet'))`);
+
+    if (commandSets.length === 0) {
+      throw `No user commandsets with ClientSideComponentId '${options.clientSideComponentId}' found`;
+    }
+    if (commandSets.length > 1) {
+      throw `Multiple user commandsets with ClientSideComponentId '${options.clientSideComponentId}' found. Please disambiguate using IDs: ${commandSets.map((commandSet: CustomAction) => commandSet.Id).join(', ')}`;
+    }
+    return commandSets[0].Id;
```

Comment:
while preparing to merge this PR https://github.com/pnp/cli-microsoft365/pull/4564
I noticed some part we may remove which is really similar to this one.
could we also try to improve this section as well?